src/components/views/rooms/RoomSettings.js

```diff
@@ -252,6 +270,30 @@ module.exports = React.createClass({
         });
     },
     
+    _onRoomAccessRadioToggle: function(ev) {
+        var self = this;
+        switch (ev.target.value) {
+            case "invite_only":
+                self.setState({
+                    join_rule: "invite",
+                    guest_access: "can_join",
```

Comment:
half the point of these new settings is to make sure that if a room is invite only, and you invite a guest, you don't have to remember to also set a 'guest access is true' checkbox.  If you're inviting someone explicitly, you know you want them to join anyway whether they're a guest or not.

So yes, you can't set all 4 states, but this is very much deliberate.  Will put the table in to spell it out.